include/my_stacktrace.h

```diff
@@ -59,6 +59,24 @@ void my_set_exception_pointers(EXCEPTION_POINTERS *ep);
 void my_write_core(int sig);
 #endif
 
+/* print stack traces for all threads */
+void my_pstack();
+
+/* print stack traces for all threads before assert failures */
+#define assert_with_stack_traces(expr) \
```

Comment:
True. MyRocks uses the abort_with_stack_traces for a SHIP_ASSERT or an assertion that is compiled into release builds. Presumably there not a concern or an issue with stack space for this use case. i.e. I don't believe there is any use of this assertion if we are at or nearing end of stack. Other than that one specific issue, I do not see why we could not use this for all assertions.